pvlib/pvsystem.py

```diff
@@ -852,6 +852,67 @@ def i_from_v(self, resistance_shunt, resistance_series, nNsVth, voltage,
         return i_from_v(resistance_shunt, resistance_series, nNsVth, voltage,
                         saturation_current, photocurrent)
 
+    def get_ac(self, model, p_dc, v_dc=None):
+        r"""Calculates AC power from p_dc using the inverter model indicated
+        by model and self.inverter_parameters.
+
+        Parameters
+        ----------
+        model : str
+            Must be one of 'sandia', 'adr', or 'pvwatts'.
+        p_dc : numeric, or tuple, list or array of numeric
+            DC power on each MPPT input of the inverter. Use tuple, list or
+            array for inverters with multiple MPPT inputs. If type is array,
+            p_dc must be 2d with axis 0 being the MPPT inputs. [W]
+        v_dc : numeric, or tuple, list or array of numeric
+            DC voltage on each MPPT input of the inverter. Required when
+            model='sandia' or model='adr'. Use tuple, list or
+            array for inverters with multiple MPPT inputs. If type is array,
+            v_dc must be 2d with axis 0 being the MPPT inputs. [V]
+
+        Returns
+        -------
+        power_ac : numeric, or tuple of numeric
+            AC power output for the inverter. [W]
+
+        Raises
+        ------
+        ValueError
+            If model is not one of 'sandia', 'adr' or 'pvwatts'.
+
+        See also
+        --------
+        pvlib.inverter.sandia
+        pvlib.inverter.sandia_multi
+        pvlib.inverter.adr
+        pvlib.inverter.pvwatts
+        pvlib.inverter.pvwatts_multi
+        """
+        model = model.lower()
+        if model == 'sandia':
+            if self.num_arrays > 1:
+                p_dc = self._validate_per_array(p_dc)
+                v_dc = self._validate_per_array(v_dc)
+                inv_fun = inverter.sandia_multi
+            else:
+                inv_fun = inverter.sandia
+            return inv_fun(v_dc, p_dc, self.inverter_parameters)
+        elif model == 'pvwatts':
+            kwargs = _build_kwargs(['eta_inv_nom', 'eta_inv_ref'],
+                                   self.inverter_parameters)
+            if self.num_arrays > 1:
+                p_dc = self._validate_per_array(p_dc)
+                inv_fun = inverter.pvwatts_multi
+            else:
+                inv_fun = inverter.pvwatts
+            return inv_fun(p_dc, self.inverter_parameters['pdc0'], **kwargs)
+        elif model == 'adr':
```

Comment:
we need to handle the case that `model == 'adr' and self.num_arrays > 1` with another `ValueError`.